include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -381,7 +387,7 @@ compared to 1*64 when the hasLeadingOffset is false.
           llvm_unreachable("unsupported shared memory layout for MMAv3");
         }
 
-        return $_get(context, vec, perPhase, maxPhase, order, CTALayout, true);
+        return $_get(context, vec, perPhase, maxPhase, order, CTALayout, hasLeadingOffset);
```

Comment:
should we assert that `hasLeadingOffset` is true? I'm not sure this function is valid for other cases

Reply:
This won't be ideal as well, since for example for 2D LUT case we will end up in this function with hasLeadingOffset == false.
This is why in my initial PR I have been checking if the loads feeds in to dot, and not if it is 1D to select unswizzled layout. I don't think we have a function that selects optimal shared layout for 2D tensors that are not used by dot.
Do you think we could adapt this function to non MMAv3 cases?

Reply:
After offline sync, we decided with @ThomasRaoux to revert to previous implementation that was using unswizzled layouts for indirect loads for now.